### PR TITLE
fix(api): Explicitly generate metadata for `nest build` script

### PR DESCRIPTION
**Pull request by rifont**
### What changed? Why was the change needed?

* Explicitly generate metadata for `nest build` script
  * The `nest build` script used in CI does not generate metadata in the same way that `nest start --watch` does during local development, therefore we need to explicitly produce the metadata using a new `generate-metadata.ts` script which runs prior to the `nest build` invocation

### Screenshots

_Running the `pnpm build` script in API_
```bash
> pb

> @novu/api@2.0.0-canary.0 prebuild /Users/rifont/git/novu/apps/api
> rimraf dist


> @novu/api@2.0.0-canary.0 build /Users/rifont/git/novu/apps/api
> pnpm build:metadata && nest build


> @novu/api@2.0.0-canary.0 build:metadata /Users/rifont/git/novu/apps/api
> cross-env ts-node scripts/generate-metadata.ts

metadata.ts file has been generated with default content.
⠋  TSC  Initializing type checker...>  TSC  Found 0 issues. Generating metadata...
✔  TSC  Initializing type checker...
✔  TSC  Initializing type checker...
>  TSC  Found 0 issues. Generating metadata...
>  SWC  Running...
Successfully compiled: 941 files with swc (406.47ms)
```

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Special notes for your reviewer

</details>

### CODE REVIEW

File: apps/api/src/metadata.ts
Comment (rifont) [inline apps/api/src/metadata.ts:1]: `metadata.ts` is no longer checked into source control and is generated by the `build` script as a pre-requisite step.

File: apps/api/scripts/generate-metadata.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ * We create an empty metadata file to ensure that files importing `metadata.ts`
+ * will compile successfully before the metadata generation occurs.
+ */
+const defaultContent = `export default async () => { return {}; };`;
+
+fs.writeFileSync(metadataPath, defaultContent, 'utf8');
+console.log('metadata.ts file has been generated with default content.');
+
+/**
+ * This file is responsible for generating Nest.js metadata for the API.
```
Comment (denis-kralj-novu) [inline apps/api/scripts/generate-metadata.ts]: 🤓 **nitpick (non-blocking):** ‏
```suggestion
 * This code snippet is responsible for generating Nest.js metadata for the API.
```

Comment (denis-kralj-novu) [reply apps/api/scripts/generate-metadata.ts]: Alternatively, move the entire comment block to the top of the file

**Review by denis-kralj-novu — APPROVED** [review]
Small comment, otherwise 🚢 IT :D

File: apps/api/scripts/generate-metadata.ts (hunk 1 above)
Comment (rifont) [reply apps/api/scripts/generate-metadata.ts]: Good call - I moved it to the top of the file.

**Pull request merged**